internal/command/views/json/diagnostic.go

```diff
@@ -499,6 +499,10 @@ func compactValueStr(val cty.Value) string {
 		return "(sensitive value)"
 	}
 
+	// val could have some marks other than sensitive ones, so we want to
+	// unmark it first to eliminate the risk of panics.
+	val, _ = val.UnmarkDeep()
```

Comment:
Hmm, currently this results in panics if the below logic has bugs in regards to sensitivity. With this change, that guarantee is no longer present.  This also may not work as intended on nested objects.

I'd prefer to filter out the deprecated marks here instead.

Reply:
Makes sense, I changed it to remove only deprecated marks in [7ada323](https://github.com/opentofu/opentofu/pull/2633/commits/7ada323bce62fa7fc55a87b402a1447eacd4fb38).